QappTuning.py

```diff
@@ -0,0 +1,585 @@
+#!/usr/bin/python3
+import os
+import os.path
+import shutil
+import sys
+import warnings
+from concurrent.futures import ThreadPoolExecutor
+
+import cv2
+import numpy as np
+from picamera2 import Picamera2, Platform
+from picamera2.previews.qt import QGlPicamera2
+from PyQt5.QtCore import QTimer, pyqtSignal
+from PyQt5.QtWidgets import (QApplication, QHBoxLayout, QLabel, QLineEdit,
+                             QListWidget, QListWidgetItem, QMainWindow,
+                             QMessageBox, QPushButton, QTabWidget, QVBoxLayout,
+                             QWidget)
+
+# finding ctt folder
+ctt_directory = os.path.join(os.path.expanduser("~"), 'libcamera/utils/raspberrypi/ctt')
+sys.path.insert(1, ctt_directory)
+from ctt import Camera, run_ctt
+from ctt_image_load import dng_load_image
+from ctt_macbeth_locator import find_macbeth, get_macbeth_chart
+
+
+# doesnt let runtime warning print
+def fxn():
+    warnings.warn("runtime", RuntimeWarning)
+
+
+# Configuring the camera
+# sorting functions for camera modes
+def sortFn(d):
+    return d['crop_limits'][-1] * d['crop_limits'][-2]
+
+
+def sortBitDepth(d):
+    return d['bit_depth']
+
+
+picam2 = Picamera2()
+candidate_modes = []
+
+
+for x in picam2.sensor_modes:
+    if 0 in x['crop_limits']:
+        candidate_modes.append(x)
+
+
+if len(candidate_modes) == 1:
+    mode = candidate_modes[0]
+else:
+    candidate_modes.sort(key=sortFn)
+    tuple_ful = candidate_modes[-1]['size']
+    tuple_half = tuple(ti / 2 for ti in tuple_ful)
+
+    half_candidates = [x for x in candidate_modes if x['size'] == tuple_half]
+    full_candidates = [x for x in candidate_modes if x['size'] == tuple_ful]
+
+    if half_candidates != []:
+        half_candidates.sort(key=sortBitDepth)
+        mode = half_candidates[-1]
+    else:
+        full_candidates.sort(key=sortBitDepth)
+        mode = full_candidates[-1]
+main = {"size": (800, 600)}
+sensor = {'output_size': mode['size'], 'bit_depth': mode['bit_depth']}
+config = picam2.create_preview_configuration(main=main, sensor=sensor)
+picam2.configure(config)
+
+
+# Function that returns true if it can detect macbeth plot
+def check(adress, target):
+    Cam = Camera("imx3.json", json=json_template)
```

Comment:
Adress fixed
Camera needs a json file (name) as one of the arguments

Reply:
Does it do anything with the name? If it really doesn't matter what we call it, then maybe `"dummy_name"` or something like that would be better. It would save people getting confused and wondering why we're talking about Sony's imx3 sensor (if such a thing even exists!).